derive_builder_core/src/try_from.rs

```diff
@@ -0,0 +1,174 @@
+use quote::{Tokens, ToTokens};
+use syn;
+
+use {Bindings, BuilderPattern};
+
+/// Generator for `TryFrom<FooBuilder>`.
+pub struct TryFromImpl<'a> {
+    /// Enables code generation for this TryFrom impl.
+    pub enabled: bool,
+    /// Name of the builder type.
+    pub builder_ty: &'a syn::Ident,
+    /// Name of the build fn.
+    pub fn_ident: syn::Ident,
+    /// How the build method takes and returns `self` (e.g. mutably).
+    pub pattern: BuilderPattern,
+    /// Type of the target field.
+    ///
+    /// The corresonding builder field will be `Option<field_type>`.
+    pub target_ty: &'a syn::Ident,
+    /// Type parameters and lifetimes attached to this builder struct.
+    pub generics: &'a syn::Generics,
+    
+    /// Bindings to libstd or libcore.
+    pub bindings: Bindings,
+}
+
+impl<'a> ToTokens for TryFromImpl<'a> {
+    fn to_tokens(&self, tokens: &mut Tokens) {
+        if !self.enabled {
+            return;
+        }
+        
+        // For the generic declarations after impl, we need a
+        // lifetime so that we can declare TryFrom<&'whatever FooBuilder>.
+        // This should NOT appear in the where clause or target type declaration,
+        // which is why we have to make the clone.
+        let mut i_generics = self.generics.clone();
+        i_generics.lifetimes.push(syn::LifetimeDef::new("'unused"));
+        let (impl_generics, _, _) = i_generics.split_for_impl();
+        
+        let builder_ty = &self.builder_ty;
+        let target_ty = &self.target_ty;
+        let (owned_generics, ty_generics, where_clause) = self.generics.split_for_impl();
+        let fn_ident = &self.fn_ident;
+        let try_from = self.bindings.try_from_trait();
+        let result_ty = self.bindings.result_ty();
+        let string_ty = self.bindings.string_ty();
+        
+        let immutable_borrow = quote!(
+            impl #impl_generics #try_from<&'unused #builder_ty #ty_generics> for #target_ty #ty_generics 
```

Comment:
What happens if we elide the lifetime parameter `'unused` and write `& #builder_ty #ty_generics` instead?
Would that fail if `#ty_generics` contained other lifetime parameters? Did you test that?

If we really need to have an explicit lifetime parameter, I'd suggest to call it `'try_from` just in case so the user would get nicer error messages - while still reducing the risk of name collisions.